crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -271,7 +271,7 @@ impl CommittedState {
         // Insert the columns into `st_columns`
         let st_columns = self.get_or_create_table(ST_COLUMNS_ID, st_columns_schema());
 
-        for col in system_tables().into_iter().flat_map(|x| x.columns) {
+        for col in system_tables().into_iter().flat_map(|x| x.columns.clone()) {
```

Comment:
Rust complains:

```
error[E0507]: cannot move out of dereference of `spacetimedb_sats::db::def::TableSchema`
   --> crates/core/src/db/datastore/locking_tx_datastore/mod.rs:274:61
    |
274 |         for col in system_tables().into_iter().flat_map(|x| x.columns) {
    |                                                             ^^^^^^^^^ move occurs because value has type `std::vec::Vec<spacetimedb_sats::db::def::ColumnSchema>`, which does not implement the `Copy` trait
```

I think this is a consequence of the `read_only` mark.

Reply:
The issue is that caching the `row_type` means any change to `columns` invalidates that, and I need a way to protect against that scenario.